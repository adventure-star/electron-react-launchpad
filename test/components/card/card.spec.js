import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import Card from '../../../src/renderer/components/card/card.jsx'

function setupStandard() {
    const renderer = ReactTestRenderer.create(<Card top-border-color='red' className='testClass'>Test card content</Card>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Card component', () => {
    it('should be of type div', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('div');
    });

    it('should have only the classes "card" and "testClass"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('card testClass');
    });

    it('children should be the text "Test card content"', () => {
        const { component } = setupStandard();
        expect(component.children[0]).toBe('Test card content');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
