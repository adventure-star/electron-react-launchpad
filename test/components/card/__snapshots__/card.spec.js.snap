// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Matches/Renders a snapshot 1`] = `
<div
  className="card testClass"
  style={
    Object {
      "borderTop": "3px solid red",
    }
  }
>
  Test card content
</div>
`;
